File: webapptoolsA.Server/Entities/Transaction.cs


namespace webapptoolsA.Server.Entities
{
    public class TransactionHeader
    {
        public int? Id { get; set; }
        public int UserId { get; set; }
        public int? Days { get; set; }
        public int IdType { get; set; }
        public string? Notes { get; set; }
        public int UserRecipt { get; set; }
        public int? IdProject { get; set; }
        public int IdWarehouseOrigin { get; set; }
        public int IdWarehouseDestination { get; set; }
        public DateTime? CreatedAt { get; set; }
        public int Status { get; set; }

        public virtual User? User { get; set; }

        public virtual User? UserReciptNavigation { get; set; }

        public virtual Warehouse? WarehouseOrigin { get; set; }

        public virtual Warehouse? WarehouseDestination { get; set; }

        public virtual TypeTransaction? Type { get; set; }

        public virtual Project? Project { get; set; }

        public virtual ICollection<TransactionDetail>? Details { get; set; }
    }

    public class TransactionDetail
    {
        public int IdDetailTransaction { get; set; }
        public int IdTransaction { get; set; }
        public int IdWarehouse { get; set; }
        public int Quantity { get; set; }

        public virtual TransactionHeader? Transaction { get; set; }
        public virtual Warehouse? Warehouse { get; set; }

    }

    public class TypeTransaction
    {
        public int? Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Type { get; set; } 
    }

    public class Project
    {
        public int? Id { get; set; }
        public string Name { get; set; } 
        public string? Description { get; set; }
        public string? Location { get; set; }
        public int UserId { get; set; }
        public DateTime? CreatedAt { get; set; }
        public virtual User? User { get; set; }
    }


}

File: webapptoolsa.client/src/app/models/Transaction.ts

export interface TypeTransaction {
  id: number | null;
  code: string;
  name: string;
  type: string;
  isActived: boolean;
}

export interface Project {
  id: number | null;
  name: string;
  description: string | null;
  location: string | null;
  userId: number;
  createdAt: Date | null;
  username: string | null;
}


export interface TransactionHeaderBase {
  id: number | null;
  userId: number;
  dateStart: Date | null;
  dateEnd: Date | null;
  days: number | null;
  idType: number;
  notes: string | null;
  userRecipt: number|null;
  idProject: number | null;
  idWarehouseOrigin: number;
  idWarehouseDestination: number | null;
  createdAt: Date | null;  
  status: number;
}

export interface TransactionHeaderDto extends TransactionHeaderBase {
  usernameRegister: string;
  usernameRecipt: string;
  warehouseOrigin: string;
  warehouseDestination: string;
  nameTypeTransaction: string;
  nameProject: string;
}

export interface TransactionDetailBaseDto {
  idDetailTransaction: number;  
  idTransaction: number;
  idWarehouse: number;
  idToolsType: number;
  idStatusTools: number;
  quantity: number;
}

export interface TransactionDetail extends TransactionDetailBaseDto {
  toolTypeName: string;
  statusToolName: string;
}

export type TransactionDetailDisplayTable = Pick<TransactionDetail, 'idToolsType'|'toolTypeName' |'quantity'>


